packages/launchpad/src/use-tracking/test/index.tsx

```diff
@@ -0,0 +1,109 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { renderHook } from '@testing-library/react';
+import { useTracking } from '..';
+import { buildTask } from '../../test/lib/fixtures';
+import type { SiteDetails } from '@automattic/data-stores';
+
+jest.mock( '@automattic/calypso-analytics' );
+
+describe( 'useTracking', () => {
+	beforeEach( () => jest.clearAllMocks() );
+
+	const buildDefaultProps = ( options = {} ) => ( {
+		checklistSlug: 'site-setup-checklist',
+		context: 'customer-home',
+		site: { options: { site_intent: 'build' } } as SiteDetails,
+		tasks: [
+			buildTask( { id: 'task-1', completed: true } ),
+			buildTask( { id: 'task-2', completed: false } ),
+		],
+		...options,
+	} );
+
+	it( 'tracks the view event on load', () => {
+		renderHook( () => useTracking( buildDefaultProps() ) );
+
+		expect( recordTracksEvent ).toHaveBeenCalledWith( 'calypso_launchpad_tasklist_viewed', {
+			checklist_slug: 'site-setup-checklist',
+			tasks: ',task-1,task-2,',
+			is_completed: false,
+			number_of_steps: 2,
+			number_of_completed_steps: 1,
+			context: 'customer-home',
+			site_intent: 'build',
+		} );
+	} );
+
+	it( 'use comma as prefix and suffix for the task names to simplify queries on the logs tool', () => {
+		renderHook( () => useTracking( buildDefaultProps() ) );
+
+		const tasks = jest.mocked( recordTracksEvent ).mock.calls[ 0 ][ 1 ].tasks;
+		expect( tasks ).toBe( ',task-1,task-2,' );
+	} );
+
+	it( 'tracks the view event by task', () => {
+		const tasks = [
+			buildTask( { id: 'task-1', completed: true, order: 3 } ),
+			buildTask( { id: 'task-2', completed: false, order: 4 } ),
+		];
+
+		renderHook( () => useTracking( buildDefaultProps( { tasks } ) ) );
+
+		expect( recordTracksEvent ).toHaveBeenNthCalledWith( 2, 'calypso_launchpad_task_view', {
+			checklist_slug: 'site-setup-checklist',
+			is_completed: true,
+			context: 'customer-home',
+			site_intent: 'build',
+			order: 3,
+			task_id: 'task-1',
+		} );
```

Comment:
Minor note: is it worth checking the call for the second task as well?
```suggestion
		} );
		
		expect( recordTracksEvent ).toHaveBeenNthCalledWith( 3, 'calypso_launchpad_task_view', {
			checklist_slug: 'site-setup-checklist',
			is_completed: false,
			context: 'customer-home',
			site_intent: 'build',
			order: 4,
			task_id: 'task-2',
		} );
```

Reply:
I added a new expectation to cover it [1b77c25](https://github.com/Automattic/wp-calypso/pull/87141/commits/1b77c25578972c8e1d6876588f6b38201b9fb60f)